task_2/task_2/BinaryHeap.cs

```diff
@@ -0,0 +1,50 @@
+﻿public class BinaryHeap<T> where T : IComparable<T>
```

Comment:
```suggestion
﻿public class BinaryHeap<T>
```
Ты сравниваешь приоритеты, а не сами элементы.
Поэтому для типа элементов `IComparable` не нужен.